android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java

```diff
@@ -302,7 +304,68 @@ public TiListView(TiViewProxy proxy, Activity activity) {
 		listView.setFocusable(true);
 		listView.setFocusableInTouchMode(true);
 		listView.setDescendantFocusability(ViewGroup.FOCUS_AFTER_DESCENDANTS);
+		final KrollProxy fProxy = proxy;
+		listView.setOnScrollListener(new OnScrollListener()
+		{
+			private int _firstVisibleItem = 0;
+			private int _visibleItemCount = 0;
+			// private int _totalItemCount = 0;
+			@Override
+			public void onScrollStateChanged(AbsListView view, int scrollState)
+			{
+				switch(scrollState) {
+
+					/*
+					 * Not supporting this one for now
+					 *
+					 * case OnScrollListener.SCROLL_STATE_FLING:
+					 * {
+					 * 	// The user had previously been scrolling using touch and had performed a fling.
+					 * 	// The animation is now coasting to a stop
+					 * 	KrollDict eventArgs = new KrollDict();
+					 * 	fProxy.fireEvent("fling", eventArgs);
+					 * 	return;
+					 * }
+					 */
+
+					case OnScrollListener.SCROLL_STATE_IDLE:
+					case OnScrollListener.SCROLL_STATE_TOUCH_SCROLL:
+					{
+						KrollDict eventArgs = new KrollDict();
+						Pair<ListSectionProxy, Pair<Integer, Integer>> info = getSectionInfoByEntryIndex(_firstVisibleItem);
+						
+						
+						int itemIndex = info.second.second;
+						ListSectionProxy section = info.first;
+						
+						if(section.getHeaderTitle() == null || section.getHeaderView() == null) {
+							if(itemIndex > 0) itemIndex -= 1;
+							_visibleItemCount -=1;
```

Comment:
This line might be causing this problem:- 

For edge case, when you are either at the top of the list and continue scrolling up or at the bottom of the list and scrolling down, it continues to count and go negative.

```
03-29 23:30:38.900: I/TiAPI(9211):  END firstVisibleSectionIndex: 29 firstVisibleItemIndex: 11 firstVisibleSection: [object ListSection] firstVisibleItem: [object Object] visibleItemCount: -16
```